src/libraries/Common/tests/System/IO/VirtualDriveHelper.Windows.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+public class VirtualDriveHelper : IDisposable
+{
+    // Temporary Windows directory that can be mounted to a drive letter using the subst command
+    private string? _virtualDriveTargetDir = null;
+    // Windows drive letter that points to a mounted directory using the subst command
+    private char _virtualDriveLetter = default;
+
+    /// <summary>
+    /// If there is a SUBST'ed drive, Dispose unmounts it to free the drive letter.
+    /// </summary>
+    public void Dispose()
+    {
+        try
+        {
+            if (VirtualDriveLetter != default)
+            {
+                DeleteVirtualDrive(VirtualDriveLetter);
+                Directory.Delete(VirtualDriveTargetDir, recursive: true);
+            }
+        }
+        catch { } // avoid exceptions on dispose
+    }
+
+    /// <summary>
+    /// Returns the path of a folder that is to be mounted using SUBST.
+    /// </summary>
+    public string VirtualDriveTargetDir
+    {
+        get
+        {
+            if (_virtualDriveTargetDir == null)
+            {
+                // Create a folder inside the temp directory so that it can be mounted to a drive letter with subst
+                _virtualDriveTargetDir = Path.Join(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(_virtualDriveTargetDir);
+            }
+
+            return _virtualDriveTargetDir;
+        }
+    }
+
+    /// <summary>
+    /// Returns the drive letter of a drive letter that represents a mounted folder using SUBST.
+    /// </summary>
+    public char VirtualDriveLetter
+    {
+        get
+        {
+            if (_virtualDriveLetter == default)
+            {
+                // Mount the folder to a drive letter
+                _virtualDriveLetter = CreateVirtualDrive(VirtualDriveTargetDir);
+            }
+            return _virtualDriveLetter;
+        }
+    }
+
+    ///<summary>
+    /// On Windows, mounts a folder to an assigned virtual drive letter using the subst command.
+    /// subst is not available in Windows Nano.
+    /// </summary>
+    public char CreateVirtualDrive(string targetDir)
+    {
+        if (!OperatingSystem.IsWindows())
```

Comment:
Since this code is stored in `.Windows.cs` file I would skip the OS checks or just apply `[SupportedOSPlatform("windows")]` attribute to the entire class